### PR TITLE
Issue #1061 - Update time and date-time format definitions to support…

### PR DESCRIPTION
**Pull request by cjpillsbury**
… two digit and colon-less variants of timezone offset. Add tests. Update test eslint to include global after function.

**What issue does this pull request resolve?**

**What changes did you make?**

**Is there anything that requires more attention while reviewing?**

### CODE REVIEW

**Review by epoberezkin — CHANGES_REQUESTED** [review]
Thank you for the PR - RFC3339 indeed allows such time-zone offsets.
It's best to add tests in JSON format in the same way [json-schema-tests-suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite) tests are defined - they can be added [here](https://github.com/epoberezkin/ajv/tree/master/spec/tests/issues) as a single file.

It is also worth submitting these tests to [json-schema-tests-suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite).

File: spec/issues/1061_alternative_time_offsets.spec.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
+"use strict";
+
+var Ajv = require("../ajv");
+require("../chai").should();
+
+describe("PR #617, full date format validation should understand leap years", function() {
```
Comment (epoberezkin) [inline spec/issues/1061_alternative_time_offsets.spec.js]: Also it is probably copied from another test

File: spec/issues/1061_alternative_time_offsets.spec.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  describe("time formats - fast", function() {
+    var ajv;
+    var schema;
+
+    before(function() {
+      ajv = new Ajv({ format: "fast" });
```
Comment (epoberezkin) [inline spec/issues/1061_alternative_time_offsets.spec.js]: JSON tests would run for both "full" and "fast" format definitions (I think)

File: spec/issues/1061_alternative_time_offsets.spec.js (hunk 1 above)
Comment (cjpillsbury) [reply spec/issues/1061_alternative_time_offsets.spec.js]: Yup definitely started by copying to better ensure formatting/code consistency. File's now gone.

File: spec/issues/1061_alternative_time_offsets.spec.js (hunk 2 above)
Comment (cjpillsbury) [reply spec/issues/1061_alternative_time_offsets.spec.js]: Confirmed that the permutations of Ajv instances generated and used by `schema-tests.spec.js` explicitly includes `format: 'full'` and implicitly (via option default) includes `format: 'fast'`.

**Comment by cjpillsbury** [comment]
> It is also worth submitting these tests to [json-schema-tests-suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite).

Cool good call I'll be sure to follow up with a PR to their repo as well. If/when that gets added, do you want me to give a heads up or issue a followup PR to remove the redundant tests from your repo?

**Comment by epoberezkin** [comment]
> do you want me to give a heads up or issue a followup PR to remove the redundant tests from your repo?

There are a few others - worth addressing together, if ever.

I am not that worried about having some overlap to be honest.

**Comment by epoberezkin** [comment]
... and thank you for updating the tests!

**Pull request merged**